packages/contracts/src/contracts/current/protocol/AssetProxyOwner/AssetProxyOwner.sol

```diff
@@ -0,0 +1,110 @@
+
+*/
+
+pragma solidity ^0.4.10;
+
+import "../../multisig/MultiSigWalletWithTimeLock.sol";
+import "../../utils/LibBytes/LibBytes.sol";
+
+contract AssetProxyOwner is
+    LibBytes,
+    MultiSigWalletWithTimeLock
+{
+
+    event AssetProxyRegistration(address assetProxyContract, bool isRegistered);
+
+    // Mapping of AssetProxy contract address => approved to execute removeAuthorizedAddress without time lock.
+    mapping (address => bool) public isAssetProxyRegistered;
+
+    bytes4 constant REMOVE_AUTHORIZED_ADDRESS_SELECTOR = bytes4(keccak256("removeAuthorizedAddress(address)"));
+
+    /// @dev Function will revert if the transaction does not call `removeAuthorizedAddress`
+    ///      on an approved AssetProxy contract.
+    modifier validRemoveAuthorizedAddressTx(uint256 transactionId) {
+        Transaction storage tx = transactions[transactionId];
+        require(isAssetProxyRegistered[tx.destination]);
+        require(isFunctionRemoveAuthorizedAddress(tx.data));
+        _;
+    }
+
+    /// @dev Contract constructor sets initial owners, required number of confirmations,
+    ///      time lock, and list of AssetProxy addresses.
+    /// @param _owners List of initial owners.
+    /// @param _assetProxyContracts Array of AssetProxy contract addresses.
+    /// @param _required Number of required confirmations.
+    /// @param _secondsTimeLocked Duration needed after a transaction is confirmed and before it becomes executable, in seconds.
+    function AssetProxyOwner(
+        address[] memory _owners,
+        address[] memory _assetProxyContracts,
+        uint256 _required,
+        uint256 _secondsTimeLocked
+    )
+        public
+        MultiSigWalletWithTimeLock(_owners, _required, _secondsTimeLocked)
+    {
+        for (uint256 i = 0; i < _assetProxyContracts.length; i++) {
+            address assetProxy = _assetProxyContracts[i];
+            require(assetProxy != address(0));
+            isAssetProxyRegistered[assetProxy] = true;
+        }
+    }
+
+    /// @dev Registers or deregisters an AssetProxy to be able to execute
+    ///      removeAuthorizedAddress without a timelock.
+    /// @param assetProxyContract Address of AssetProxy contract.
+    /// @param isRegistered Status of approval for AssetProxy contract.
+    function registerAssetProxy(address assetProxyContract, bool isRegistered)
+        public
+        onlyWallet
+        notNull(assetProxyContract)
+    {
+        isAssetProxyRegistered[assetProxyContract] = isRegistered;
+        AssetProxyRegistration(assetProxyContract, isRegistered);
+    }
+
+    /// @dev Allows execution of removeAuthorizedAddress without time lock.
+    /// @param transactionId Transaction ID.
+    function executeRemoveAuthorizedAddress(uint256 transactionId)
+        public
+        notExecuted(transactionId)
+        fullyConfirmed(transactionId)
+        validRemoveAuthorizedAddressTx(transactionId)
+    {
+        Transaction storage tx = transactions[transactionId];
```

Comment:
Can this just call the MultiSigWallet `executeTransaction` after the additional modifier validation?

Reply:
We can't because we're trying to bypass the `pastTimeLock` modifier.

Reply:
I guess we're adverse to refactoring to `executeTransactionInternal` and a public executeTransaction with the modifiers? Mostly due to the fact that these contracts have already been audited?